tests/ops/test_hamiltonian.py

```diff
@@ -762,311 +769,109 @@ def test_simplify(self, coeffs):
-
-        coeffs2 = torch.tensor([0.5, -0.4])
-        H2 = qml.Hamiltonian(coeffs2, obs)
-
-        coeffs_expected = torch.tensor([0.5, -1.6])
-        H = qml.Hamiltonian(coeffs_expected, obs)
-
-        assert H.compare(H1 - H2)
-
-        H1 -= H2
-        assert H.compare(H1)
-
-    def test_hamiltonian_matmul(self):
-        """Tests that Hamiltonians are tensored correctly"""
-        torch = pytest.importorskip("torch")
-
-        coeffs = torch.tensor([1.0, 2.0])
-        obs = [qml.PauliX(0), qml.PauliY(1)]
-        H1 = qml.Hamiltonian(coeffs, obs)
-
-        coeffs2 = torch.tensor([-1.0, -2.0])
-        obs2 = [qml.PauliX(2), qml.PauliY(3)]
-        H2 = qml.Hamiltonian(coeffs2, obs2)
-
-        coeffs_expected = torch.tensor([-4.0, -2.0, -2.0, -1.0])
-        obs_expected = [
-            qml.PauliY(1) @ qml.PauliY(3),
-            qml.PauliX(0) @ qml.PauliY(3),
-            qml.PauliX(2) @ qml.PauliY(1),
-            qml.PauliX(0) @ qml.PauliX(2),
-        ]
-        H = qml.Hamiltonian(coeffs_expected, obs_expected)
-
-        assert H.compare(H1 @ H2)
-
-
-class TestHamiltonianArithmeticAutograd:
-    """Tests creation of Hamiltonians using arithmetic
-    operations with autograd tensor coefficients."""
-
-    def test_hamiltonian_equal(self):
+class TestSparseRepresentation:
+    """Tests the sparse representation of a Hamiltonian"""
+
+    @pytest.mark.parametrize("converter", converters)
+    def test_no_wires(self, converter):
+        """Test representation using the Hamiltonian's wires"""
+        ham = qml.Hamiltonian(converter([0.1, 0.2]), [qml.PauliX('b'), qml.PauliY('a')])
+        s = ham.sparse_matrix()
+        expected = {(0, 2): 0.1, (1, 3): 0.1, (2, 0):0.1, (3, 1): 0.1,
+                    (0, 1): 0.-0.2j, (1, 0): 0.+0.2j, (2, 3): 0.-0.2j, (3, 2): 0.+0.2j}
+        assert s.data == {k: converter(v) for k, v in expected.items()}
+        assert s.shape == (4, 4)
+
+    @pytest.mark.parametrize("converter", converters)
+    def test_no_wires(self, converter):
+        """Test representation using custom wires"""
+        ham = qml.Hamiltonian(converter([0.1]), [qml.PauliX('b')])
+        s = ham.sparse_matrix(wires=['a', 'b'])
+        expected = {(0, 1): 0.1, (1, 0): 0.1, (2, 3): 0.1, (3, 2): 0.1}
+        assert s.data == {k: converter(v) for k, v in expected.items()}
+        assert s.shape == (4, 4)
+
+    def test_empty_hamiltonian(self):
+        """Test representation of an empty Hamiltonian"""
+        ham = qml.Hamiltonian([], [])
+        s = ham.sparse_matrix()
+        assert s.data == {}
+        assert s.shape == (1, 1)
+
+    def test_error_two_qubit_ops(self):
+        """Test that error is raised when Hamiltonian is not constructed from 1-qubit ops only."""
+        ham = qml.Hamiltonian([0.1], [qml.Hermitian(np.eye(4), wires=[0, 2])@qml.PauliX(1)])
+        with pytest.raises(ValueError, match="Can only compute sparse"):
+            ham.sparse_matrix()
+
+
+class TestHamiltonianArithmetic:
```

Comment:
Another cosmetic change: I realised that I can use `converter` to summarise the Arithmetic test classes!